controllers/keda/cloudevent_controller.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2023 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keda
+
+import (
+	"context"
+	"sync"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	"github.com/kedacore/keda/v2/pkg/eventemitter"
+	"github.com/kedacore/keda/v2/pkg/prommetrics"
+)
+
+// CloudEventsReconciler reconciles a CloudEvents object
+type CloudEventsReconciler struct {
+	client.Client
+	EventEmitter eventemitter.EventEmitter
+
+	cloudEventsGenerations *sync.Map
+}
+
+type cloudEventsMetricsData struct {
+	namespace string
+}
+
+var (
+	cloudEventPromMetricsMap  map[string]cloudEventsMetricsData
+	cloudEventPromMetricsLock *sync.Mutex
+)
+
+func init() {
+	cloudEventPromMetricsMap = make(map[string]cloudEventsMetricsData)
+	cloudEventPromMetricsLock = &sync.Mutex{}
+}
+
+// +kubebuilder:rbac:groups=keda.sh,resources=cloudevents;cloudevents/status,verbs="*"
```

Comment:
Can we please move this resource to a new group under `keda.sh`? `eventing.keda.sh` or similar.